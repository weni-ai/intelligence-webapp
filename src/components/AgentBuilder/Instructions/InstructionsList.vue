<template>
  <section
    :class="[
      'instructions-list',
      {
        'instructions-list--loading': isLoading,
        'instructions-list--no-instructions': instructions.length === 0,
      },
    ]"
    data-testid="instructions"
  >
    <template v-if="isLoading">
      <UnnnicSkeletonLoading
        v-for="index in 10"
        :key="index"
        tag="div"
        width="100%"
        height="60px"
        data-testid="instruction-loading"
      />
    </template>
    <section
      v-else-if="instructions.length === 0"
      class="instructions-list__no-instructions"
    >
      <UnnnicIcon
        icon="format_list_bulleted"
        size="xl"
        scheme="neutral-cleanest"
      />
      <p class="no-instructions__text">
        {{
          $t(
            'agent_builder.instructions.instructions_list.no_custom_instructions',
          )
        }}
      </p>
    </section>
    <template v-else>
      <Instruction
        v-for="instruction in instructions"
        :key="instruction.id"
        :instruction="instruction"
        :showActions="showActions"
        data-testid="instruction-added"
      />
    </template>
  </section>
</template>

<script setup>
import Instruction from './Instruction.vue';

defineProps({
  instructions: {
    type: Array,
    required: true,
  },
  isLoading: {
    type: Boolean,
    default: false,
  },
  showActions: {
    type: Boolean,
    default: false,
  },
});
</script>

<style lang="scss" scoped>
.instructions-list {
  margin-bottom: $unnnic-spacing-sm;

  display: flex;
  flex-direction: column;

  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  &--loading,
  &--no-instructions {
    height: 100%;

    gap: $unnnic-spacing-nano;

    border: none;
    border-radius: 0;
  }

  &__no-instructions {
    height: 100%;

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap: $unnnic-spacing-nano;

    .no-instructions__text {
      color: $unnnic-color-neutral-cloudy;
      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-gt;
      font-weight: $unnnic-font-weight-regular;
      line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    }
  }
}
</style>
