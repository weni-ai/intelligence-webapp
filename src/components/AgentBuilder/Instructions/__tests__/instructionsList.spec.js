import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';

import InstructionsList from '../InstructionsList.vue';

describe('InstructionsList.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(InstructionsList, {
      props: {
        instructions: [
          { id: 1, text: 'Instruction 1' },
          { id: 2, text: 'Instruction 2' },
        ],
        isLoading: false,
        showActions: false,
      },
      global: {
        stubs: ['UnnnicSkeletonLoading'],
      },
    });
  });

  const SELECTORS = {
    instructions: '[data-testid="instructions"]',
    instructionLoading: '[data-testid="instruction-loading"]',
  };

  const findComponent = (component) =>
    wrapper.findComponent(SELECTORS[component]);
  const find = (selector) => wrapper.find(SELECTORS[selector]);

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('Component rendering', () => {
    it('renders components correctly', () => {
      expect(find('instructions').exists()).toBe(true);
      expect(findComponent('instructionLoading').exists()).toBe(false);
    });
  });

  describe('Loading state', () => {
    it('renders loading elements when status is loading', async () => {
      wrapper.setProps({ isLoading: true });
      await wrapper.vm.$nextTick();
      expect(find('instructions').exists()).toBe(true);
      expect(find('instructions').classes()).toContain(
        'instructions-list--loading',
      );
      expect(
        wrapper.findAllComponents(SELECTORS.instructionLoading).length,
      ).toBe(10);
    });

    it('hides loading list when status is complete', async () => {
      wrapper.setProps({ isLoading: false });
      await wrapper.vm.$nextTick();
      expect(findComponent('instructionLoading').exists()).toBe(false);
    });
  });

  describe('No instructions state', () => {
    it('renders no instructions list when instructions are empty', async () => {
      wrapper.setProps({ instructions: [] });
      await wrapper.vm.$nextTick();
      expect(find('instructions').exists()).toBe(true);
      expect(find('instructions').classes()).toContain(
        'instructions-list--no-instructions',
      );
      expect(
        find('instructions')
          .findComponent(SELECTORS.instructionLoading)
          .exists(),
      ).toBe(false);
    });
  });
});
