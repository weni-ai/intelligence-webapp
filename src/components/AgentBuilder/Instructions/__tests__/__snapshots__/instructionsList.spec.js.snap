// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InstructionsList.vue > should match snapshot 1`] = `
"<section data-v-e11d36b5="" class="instructions-list" data-testid="instructions">
  <instruction-stub data-v-e11d36b5="" instruction="[object Object]" tag="" showactions="false" data-testid="instruction-added"></instruction-stub>
  <instruction-stub data-v-e11d36b5="" instruction="[object Object]" tag="" showactions="false" data-testid="instruction-added"></instruction-stub>
</section>"
`;
